src/Integration/Suppression/SuppressedIssuesProvider.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Integration.NewConnectedMode;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    [Export(typeof(ISonarQubeIssuesProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class SuppressedIssuesProvider : ISonarQubeIssuesProvider
+    {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly ISonarQubeIssuesProviderFactory sonarQubeIssuesProviderFactory;
+
+        private ISonarQubeIssuesProvider instance;
+        private bool disposed;
+
+        [ImportingConstructor]
+        public SuppressedIssuesProvider(ISonarQubeIssuesProviderFactory sonarQubeIssuesProviderFactory, 
```

Comment:
This is too complicated and confusing: the ultimate consumer of the issues provider is the `IssuesFilter` class. However, it doesn't MEF-import the exported `ISonarQubeIssuesProviderFactory`.
Instead, it directly MEF-imports an implementation of `ISonarQubeIssueProvider`, which in turn MEF-imports the `ISonarQubeIssuesProviderFactory`.

Reply:
FYI for the avoidance of doubt, this code doesn't work in VS in its current form. For MEF to be able to use an `ImportingConstructor` it must be able to find matching `Export`s for all of the constructor parameters.

For this code to work, you would need to add an `[Export(typeof(ISonarQubeIssuesProviderFactory))]` attribute on the factory class and an `[ImportingConstructor]` on the factory class public constructor.

When making changes to how MEF components are created and wired up, it's worth debugging the extension in VS to make sure that VS can create all of the expected components (MEF composition errors happen silently in the background so they are easy to miss).

Reply:
Yep you're right, my bad. Extracted to class and didn't re-run the extension.